packages/flutter_tools/lib/src/version.dart

```diff
@@ -96,11 +153,141 @@ class FlutterVersion {
 
     return '$branch/$commit';
   }
+
+  /// We warn the user if the age of their Flutter installation is greater than
+  /// this duration.
+  @visibleForTesting
+  static const Duration kVersionAgeConsideredUpToDate = const Duration(days: 4 * 7);
+
+  /// The amount of time we wait before pinging the server to check for the
+  /// availability of a newer version of Flutter.
+  @visibleForTesting
+  static const Duration kCheckAgeConsideredUpToDate = const Duration(days: 1);
+
+  /// The prefix of the stamp file where we cache Flutter version check data.
+  @visibleForTesting
+  static const String kFlutterVersionCheckStampFile = 'flutter_version_check';
+
+  /// Checks if the currently installed version of Flutter is up-to-date, and
+  /// warns the user if it isn't.
+  ///
+  /// This function must run while [Cache.lock] is acquired because it reads and
+  /// writes shared cache files.
+  Future<Null> checkFlutterVersionFreshness() async {
+    final DateTime localFrameworkCommitDate = DateTime.parse(frameworkCommitDate);
+    final Duration frameworkAge = clock.now().difference(localFrameworkCommitDate);
+    final bool installationSeemsOutdated = frameworkAge > kVersionAgeConsideredUpToDate;
+
+    Future<bool> newerFrameworkVersionAvailable() async {
+      final DateTime latestFlutterCommitDate = await _getLatestAvailableFlutterVersion();
+
+      if (latestFlutterCommitDate == null)
+        return false;
+
+      return latestFlutterCommitDate.isAfter(localFrameworkCommitDate);
+    }
+
+    if (installationSeemsOutdated && await newerFrameworkVersionAvailable()) {
+      printStatus(versionOutOfDateMessage(frameworkAge), emphasis: true);
+    }
+  }
+
+  @visibleForTesting
+  static String versionOutOfDateMessage(Duration frameworkAge) {
+    String warning = 'WARNING: your installation of Flutter is ${frameworkAge.inDays} days old.';
+    // Append enough spaces to match the message box width.
+    warning += ' ' * (74 - warning.length);
+
+    return '''
+  ╔════════════════════════════════════════════════════════════════════════════╗
+  ║ $warning ║
+  ║                                                                            ║
+  ║ To update to the latest version, run flutter upgrade.                      ║
```

Comment:
maybe `run "flutter upgrade".` to make it clear the `.` isn't part of the command?